src/textual/widgets/_data_table.py

```diff
@@ -408,72 +664,103 @@ def watch_show_cursor(self, show_cursor: bool) -> None:
                 self._highlight_row(self.cursor_row)
             elif self.cursor_type == "column":
                 self._highlight_column(self.cursor_column)
 
-    def watch_show_header(self, show_header: bool) -> None:
+    def watch_show_header(self) -> None:
         self._clear_caches()
 
-    def watch_fixed_rows(self, fixed_rows: int) -> None:
+    def watch_fixed_rows(self) -> None:
         self._clear_caches()
 
-    def watch_zebra_stripes(self, zebra_stripes: bool) -> None:
+    def watch_zebra_stripes(self) -> None:
         self._clear_caches()
 
-    def watch_hover_cell(self, old: Coordinate, value: Coordinate) -> None:
-        self.refresh_cell(*old)
-        self.refresh_cell(*value)
+    def watch_hover_coordinate(self, old: Coordinate, value: Coordinate) -> None:
+        self.refresh_coordinate(old)
+        self.refresh_coordinate(value)
 
-    def watch_cursor_cell(
+    def watch_cursor_coordinate(
         self, old_coordinate: Coordinate, new_coordinate: Coordinate
     ) -> None:
         if old_coordinate != new_coordinate:
             # Refresh the old and the new cell, and post the appropriate
             # message to tell users of the newly highlighted row/cell/column.
             if self.cursor_type == "cell":
-                self.refresh_cell(*old_coordinate)
-                self._highlight_cell(new_coordinate)
+                self.refresh_coordinate(old_coordinate)
+                self._highlight_coordinate(new_coordinate)
             elif self.cursor_type == "row":
                 self.refresh_row(old_coordinate.row)
                 self._highlight_row(new_coordinate.row)
             elif self.cursor_type == "column":
                 self.refresh_column(old_coordinate.column)
                 self._highlight_column(new_coordinate.column)
 
-    def _highlight_cell(self, coordinate: Coordinate) -> None:
+    def _highlight_coordinate(self, coordinate: Coordinate) -> None:
         """Apply highlighting to the cell at the coordinate, and post event."""
-        self.refresh_cell(*coordinate)
+        self.refresh_coordinate(coordinate)
         try:
-            cell_value = self.get_cell_value(coordinate)
+            cell_value = self.get_cell_at(coordinate)
         except CellDoesNotExist:
             # The cell may not exist e.g. when the table is cleared.
             # In that case, there's nothing for us to do here.
             return
         else:
+            cell_key = self.coordinate_to_cell_key(coordinate)
             self.post_message_no_wait(
-                DataTable.CellHighlighted(self, cell_value, coordinate)
+                DataTable.CellHighlighted(
+                    self, cell_value, coordinate=coordinate, cell_key=cell_key
+                )
             )
 
+    def coordinate_to_cell_key(self, coordinate: Coordinate) -> CellKey:
+        """Return the key for the cell currently occupying this coordinate in the
+        DataTable
```

Comment:
May want reformatting to confirm with [our docstring guidelines](https://github.com/Textualize/textual/wiki/Documentation-standards#class).